torch/distributed/distributed_c10d.py

```diff
@@ -1088,15 +1104,15 @@ def _new_process_group_helper(
                         timeout=timeout,
                     )
 
-        # only create single backend pg when backend is set to gloo, nccl, mpi, and ucc
-        if backend in [Backend.GLOO, Backend.NCCL, Backend.UCC, Backend.MPI]:
+        # make a single backend when all get_device_backend_map values are the same
+        if set(backend_config.get_device_backend_map().values()) == {backend_str}:
```

Comment:
Maybe:
`if len(...) == 1`
to make it simpler?

Reply:
Thanks! updated